x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/action/UnfollowAction.java

```diff
@@ -76,4 +81,91 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
+    public static class Response extends AcknowledgedResponse implements ToXContentObject {
+
+        private boolean retentionLeasesRemoved;
+        @Nullable private Exception retentionLeasesRemovalFailureCause;
+
+        public Response() {
+
+        }
+
+        public Response(
+                final boolean retentionLeasesRemoved,
+                final Exception retentionLeasesRemovalFailureCause) {
+            super(true);
+            this.retentionLeasesRemoved = retentionLeasesRemoved;
+            if (retentionLeasesRemoved && retentionLeasesRemovalFailureCause != null) {
+                throw new IllegalArgumentException(
+                        "there should not be a failure cause when retention leases are removed",
+                        retentionLeasesRemovalFailureCause);
+            } else if (retentionLeasesRemoved == false && retentionLeasesRemovalFailureCause == null) {
+                throw new IllegalArgumentException("there should be a failure cause when retention leases are not removed");
+            }
+            this.retentionLeasesRemovalFailureCause = retentionLeasesRemovalFailureCause;
+        }
+
+        @Override
+        public void readFrom(final StreamInput in) throws IOException {
+            super.readFrom(in);
+            if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
+                retentionLeasesRemoved = in.readBoolean();
+                // noinspection StatementWithEmptyBody
+                if (retentionLeasesRemoved) {
+
```

Comment:
Did you try to have a symmetric flow with writeTo?

Reply:
Yes, but now I have removed this code. 😇